src/components/InputImage.vue

```diff
@@ -1,74 +1,110 @@
 <template>
-  <div>
-    <v-layout justify-center>
-      <v-flex md5>
-        <v-card>
-          <div @click="launchFilePicker()">
-            <slot name="activator"></slot>
-          </div>
-        </v-card>
-      </v-flex>
-    </v-layout>
-    <input type="file"
-    ref="image"
-    name="image"
-    @change="onFileChange(
-      $event.target.name, $event.target.files)"
-    style="display:none">
-
-    <v-dialog v-model="errorDialog" max-width="300">
-      <v-card>
-        <v-card-text class="subheading">{{errorText}}</v-card-text>
-        <v-card-actions>
-          <v-spacer></v-spacer>
-          <v-btn @click="errorDialog = false" flat>Got it!</v-btn>
-        </v-card-actions>
-      </v-card>
-    </v-dialog>
-  </div>
+  <ValidationProvider
+    :name="name"
+    :rules="validationRules"
+    v-slot="{validate}"
+    :bails="false"
+    class="d-block">
+      <div @click="launchFilePicker()">
+        <slot></slot>
+      </div>
+    <input
+      type="file"
+      ref="image"
+      :name="name"
+      @change="onFileChange($event.target.name, $event.target.files, validate)"
+      class="d-none">
+    <div class="mb-4"></div>
+    <transition-group tag="div" name="fade-transition">
+      <div v-for="(error, index) in validationErrors" :key="error">
+          <v-fade-transition>
+            <v-alert
+              :dense="alertDense"
+              :colored-border="alertColoredBorder"
+              :elevation="alertElevation"
+              :border="borderLocation"
+              :type="alertType"
+              dismissible
+              @input="deleteError(index)"
+              >
+              {{error}}
+            </v-alert>
+          </v-fade-transition>
+      </div>
+    </transition-group>
+  </ValidationProvider>
 </template>
 
 <script>
+import { ValidationProvider } from 'vee-validate'
+
 export default {
   name: 'InputImage',
+  components: {
+    ValidationProvider
+  },
+  props: {
+    name: {
+      type: String,
+      required: true
+    },
+    icon: {
+      type: String
+    },
+    value: {
+      required: true
+    },
+    validationRules: {
+      type: Object,
+      required: true
+    },
+    alertType: {
+      type: String
+    },
+    alertElevation: {
+      type: String
+    },
+    alertDense: {
+      type: Boolean
+    },
+    borderLocation: {
+      type: String
+    },
+    alertColoredBorder: {
+      type: Boolean
+    }
```

Comment:
Naming error. Use 'BorderLocation' or 'alertBorderLocation' AND 'alertDense' or 'dense'. F.e.
Redundant.

Reply:
Refactored in eca7b959561a3659a4ce282568540f26e1eb56a0